lib/reqd.ml

```diff
@@ -474,20 +498,31 @@ let flush_response_body t ~max_bytes =
       else if Body.is_closed response_body then (
         (* no pending output and closed, we can finalize the message and close
            the stream *)
-        let frame_info =
-          Writer.make_frame_info
-            ~max_frame_size:t.max_frame_size
-            ~flags:Flags.(set_end_stream default_flags)
-            t.id
+        let frame_info flags =
```

Comment:
Why does this need to be a function? it looks like you're always setting the `END_STREAM` flag below, anyway?